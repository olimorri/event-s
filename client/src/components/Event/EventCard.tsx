import React from 'react';
import { Box, Flex, Image , Stack, Heading, Text, useColorModeValue } from '@chakra-ui/react';
import { Link as RouterLink } from 'react-router-dom';
import moment from 'moment';
import { motion } from 'framer-motion';


function EventCard ({value}) {

  return (
    <Box
    maxW={'40%'}
    h={'250px'}
    maxH={'50%'}
    w={'25%'}
    bg={useColorModeValue('white', 'gray.900')}
    boxShadow={'2xl'}
    m={2}
    borderRadius="md"
    p={3}
    overflow={'hidden'}>
          <motion.div 
          animate={{
              scale: [1, 0.8, 1],
              // @ts-ignore
              duration: 0.5
            }} 
          transition={{
            duration: 0.5,
            ease: "easeInOut",
            times: [0, 0.2, 0.5, 0.8, 0],
            // loop: Infinity,
            // repeatDelay: 1
          }}
          whileHover={{
            scale: 1.1,
            transition: {
              duration: .2
            }
          }}>
          <RouterLink to={
            {
              pathname: `/events/${value._id}`,
              state: {event: value}}
            }>
              <Flex flexDirection="column" justifyContent="center" alignItems="center">

                <Box
                  h={'20%'}
                  bg={'gray.100'}
                  mt={-6}
                  mx={-6}
                  mb={2}
                  pos={'relative'}
                  >
                    <Image
                      borderRadius="md"
                      src={value.photo}
                      // layout={'fill'}
                      />
                </Box>
                <Stack>
                  <Heading
                    color={useColorModeValue('gray.700', 'white')}
                    fontSize={'medium'}
                    fontFamily={'body'}>
                    {value.name}
                  </Heading>
                  <Text color={'gray.500'}>{value.location}</Text>
                  <Text >{moment(value.date).format("MMM Do YY")}</Text>
                </Stack>

              </Flex>

        </RouterLink>
    </motion.div>
      </Box>
  )
}

export default EventCard;