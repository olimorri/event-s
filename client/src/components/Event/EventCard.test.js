import { render, cleanup, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import EventCard from './EventCard';

afterEach(cleanup);

const event = {
  attendees: 1,
  date: '2021-03-23T00:00:00.000Z',
  description: 'This is a test event',
  duration: 2,
  geolocation: '0.261773,51.116312',
  limit: 2,
  list: ['604a36891b95d669d99ebe2e'],
  location: 'tn25hs',
  name: 'test event',
  owner: '604b5c00ca29f24f751f492c',
  photo: 'https://source.unsplash.com/random/300x300',
  type: 'Food',
};

const EventCardTest = () => (
  <BrowserRouter>
    <EventCard event={event} />
  </BrowserRouter>
);

afterEach(cleanup);

describe('Snapshot', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<EventCardTest />);
    expect(asFragment(<EventCardTest />)).toMatchSnapshot();
  });
});

describe('Unit Testing', () => {
  test('renders img', () => {
    render(<EventCardTest />);
    const displayedImg = document.querySelector('img');
    expect(displayedImg).toBeValid();
  });

  test('rendered event name matches props event.name', () => {
    render(<EventCardTest />);
    const eventName = screen.getByText(event.name);
    expect(eventName).toBeInTheDocument();
  });

  test('rendered event postcode matches props event.location', () => {
    render(<EventCardTest />);
    const eventPostcode = screen.getByText(event.location);
    expect(eventPostcode).toBeInTheDocument();
  });

  test('rendered event date matches props event.date', () => {
    render(<EventCardTest />);
    const eventDate = screen.getByText(/Mar 23rd 21/i);
    expect(eventDate).toBeInTheDocument();
  });
});
