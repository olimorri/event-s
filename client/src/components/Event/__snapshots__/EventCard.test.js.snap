// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-ujugy8"
  >
    <div>
      <a
        href="/events/undefined"
      >
        <div
          class="css-u6v8er"
        >
          <div
            class="css-1bj41hx"
          >
            <img
              class="chakra-image__placeholder css-clx8i5"
            />
          </div>
          <div
            class="chakra-stack css-1pifhv5"
          >
            <h2
              class="chakra-heading css-y9ag95"
            >
              test event
            </h2>
            <p
              class="chakra-text css-ax9bdp"
            >
              tn25hs
            </p>
            <p
              class="chakra-text css-0"
            >
              Mar 23rd 21
            </p>
          </div>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
